### ISSUE TITLE
unexpected JSON config error

### ISSUE THREAD

**Issue by NeverBehave** [shadowsocks/shadowsocks-rust#830]
##### Summary

When using `-c` arg passing in the following JSON file, an unexpected error occur, while command line with same parameter works well

##### How to reproduce

```json
{
    "method": "aes-256-gcm",
    "password": "pass",
    "server_port": 8888,
    "server": "[::]"
}
```

```bash
$ ssserver -c shadowsocks.json
2022-05-03T14:59:25.811221563-07:00 INFO  shadowsocks server 1.14.2 build 2022-04-11T03:43:00.819770073+00:00
2022-05-03T14:59:25.812564846-07:00 ERROR servers exited with error: dns resolve [::]:8888 error: proto error: Label contains invalid characters: Err(Errors { invalid_mapping, disallowed_by_std3_ascii_rules })
server aborted with server exited unexpectedly
```

While the following equivalent works just fine:

```bash
$ ssserver -s "[::]:8888" -m "aes-256-gcm" -k "pass"
2022-05-03T15:03:17.690314702-07:00 INFO  shadowsocks server 1.14.2 build 2022-04-11T03:43:00.819770073+00:00
2022-05-03T15:03:17.691616534-07:00 INFO  shadowsocks tcp server listening on [::]:8888, inbound address [::]:8888 
```

##### Expected Behavior

JSON file should behave the same as the command args

**Comment by database64128** [comment]
Don't put `::` in `[]`.

**Comment by NeverBehave** [comment]
> Don't put `::` in `[]`.

If you mean by the JSON file, here is the outcome

```
2022-05-03T15:20:16.052493707-07:00 ERROR servers exited with error: dns resolve []:8888 error: proto error: Label contains invalid characters: Err(Errors { invalid_mapping, disallowed_by_std3_ascii_rules })
server aborted with server exited unexpectedly
```

**Comment by NeverBehave** [comment]
> Don't put `::` in `[]`.

https://github.com/shadowsocks/shadowsocks-rust/blob/32f6a09d546cbb9801122c571e1b8590b5ab7886/README.md?plain=1#L539
